### PR TITLE
add module osquery v1.0.0

### PR DESCRIPTION
**Pull request by 0legovich**
Closes https://github.com/vxcontrol/soldr-modules/issues/12

### CODE REVIEW

**Comment by asdek** [comment]
@0legovich please, don't forget about Dockerfile and config.json files before giving the module to review

File: osquery/1.0.0/cmodule/helpers.lua [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    local existcmd = 'if exist ' .. path .. ' (echo yes) else (echo no)'
+    local existout = exec_cmd(existcmd) or ""
+
+    return existout == "yes\n"
+end
+
+-- arguments:
+-- *  path - path to removed dir
+-- return bool
+local function remove_dir(path)
```
Comment (asdek) [inline osquery/1.0.0/cmodule/helpers.lua:113]: I think there need to use recursive remove function for clean up directory like this
https://github.com/vxcontrol/soldr-modules/blob/master/utils/mock/core.lua#L163

File: osquery/1.0.0/cmodule/helpers.lua [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    local ok = file:write(content)
+    file:close()
+
+    if not ok then
+        return false, "can't write content to file " .. path
+    end
+
+    -- copying dump
+    local cmd = "copy " .. path_to_dumped_file .. " " .. path
+    local out = exec_cmd(cmd) or ""
```
Comment (asdek) [inline osquery/1.0.0/cmodule/helpers.lua:146]: let's change `copy` call to pure lua implementation like this
https://github.com/vxcontrol/soldr-modules/blob/master/utils/mock/core.lua#L593

**Review by asdek — COMMENTED** [review]
Also you need to sign your commits

File: osquery/1.0.0/cmodule/osquery.lua [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    local out = helpers.exec_cmd(cmd)
+    __log.debugf("out of cmd '%s': %s", cmd, out)
+
+    local result = out:match("%d+.%d+.%d+") or ""
+
+    return result
+end
+
+function Osquery:get_msi_version()
+    local cmd = 'powershell -executionpolicy bypass -File ' .. get_msi_package_script_path .. ' ' .. data_osquery_msi_path
```
[the remark this answers is not in the document]
Comment (0legovich) [reply osquery/1.0.0/cmodule/osquery.lua:140]: I didn't found how we can get a version from msi package. Do you have any ideas?

[the remark this answers is not in the document]
Comment (0legovich) [reply osquery/1.0.0/cmodule/osquery.lua]: done

**Review by asdek — APPROVED** [review]

**Pull request merged**